CDCgov/prime-reportstream
# Implement code to condition mapping and remerge CLI update tool

This PR implements code to condition mapping and re-merges previously merged PR (#12327) that was mistakenly reverted as part of #12618.

Test Steps:
1. Run unit tests and integration tests
2. Submit a report and verify condition code extensions are added
   1. On `organizations.yml:114`, modify sender `development.dev-elims` to have a  `format` of `FHIR`
   2. Starting with a clean database and reloaded settings/tables, submit a report using the updated sender 
        ```zsh
        curl --location 'http://localhost:7071/api/reports' --header 'client: development.dev-elims' --header 'Content-Type: application/fhir+ndjson' --data-binary @src/test/resources/fhirengine/engine/routing/valid.fhir
        ```
   3. Open the azure storage explorer and browse to `reports/route/development.dev-elims`
   4. Download and view the most recent (and only) blob
   5. Verify there are new `condition-code` extensions on the fhir bundle. The full url is: 
        ```html
        https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code
        ```

## Changes
- Implemented Observation extension function for adding mapped conditions
- Modified FHIRConverter to add mapped conditions to all observations
- Revive changes from #12327 

## Checklist

### Testing

## Linked Issues
- Fixes #12275 
- Fixes #11510
